src/coreclr/vm/threadsuspend.cpp

```diff
@@ -3331,11 +3331,16 @@ void ThreadSuspend::SuspendRuntime(ThreadSuspend::SUSPEND_REASON reason)
                 STRESS_LOG3(LF_SYNC, LL_INFO10000, "    Inspecting thread 0x%x ID 0x%x coop mode = %d\n",
                     thread, thread->GetThreadId(), thread->m_fPreemptiveGCDisabled.LoadWithoutBarrier());
 
+
                 // ::FlushProcessWriteBuffers above guarantees that the state that we see here
                 // is after the trap flag is visible to the other thread.
                 //
                 // In other words: any threads seen in preemptive mode are no longer interesting to us.
                 // if they try switch to cooperative, they would see the flag set. 
+#ifdef FEATURE_PERFTRACING
+                // Mark that the thread is currently in managed code.
+                thread->SaveGCModeOnSuspension();
```

Comment:
`HandleSuspensionForInterruptedThread` got called for every *safe* activation injection. This only happened on threads where: 1) a GC suspension was happening and 2) the top frame was managed. This will happen in other cases (debugger, profiler, rejit, etc) and will happen regardless of safepoints/sync state/frames.

Reply:
It's going to be consistent with the value before - it's just potentially going to be seen more often. One thing that I don't know for EP is mostly by this point there's threads that might have trapped already and switched to preemptive. If that's good enough for your calculations, then this looks good.